src/main/java/com/iota/iri/IRI.java

```diff
@@ -16,314 +16,354 @@
 import java.util.Optional;
 
 /**
- * Main IOTA Reference Implementation starting class
+ * Main IOTA Reference Implementation starting class.
  */
 public class IRI {
 
-    private static final Logger log = LoggerFactory.getLogger(IRI.class);
-
     public static final String MAINNET_NAME = "IRI";
     public static final String TESTNET_NAME = "IRI Testnet";
     public static final String VERSION = "1.4.2.4";
-    public static Iota iota;
-    public static API api;
-    public static IXI ixi;
-    public static Configuration configuration;
-    private static final String TESTNET_FLAG_REQUIRED = "--testnet flag must be turned on to use ";
-
-    public static void main(final String[] args) throws IOException {
-        configuration = new Configuration();
-        validateParams(configuration, args);
-        log.info("Welcome to {} {}", configuration.booling(DefaultConfSettings.TESTNET) ? TESTNET_NAME : MAINNET_NAME, VERSION);
-        iota = new Iota(configuration);
-        ixi = new IXI(iota);
-        api = new API(iota, ixi);
-        shutdownHook();
-
-        if (configuration.booling(DefaultConfSettings.DEBUG)) {
-            log.info("You have set the debug flag. To enable debug output, you need to uncomment the DEBUG appender in the source tree at iri/src/main/resources/logback.xml and re-package iri.jar");
+
+    public static void main(String[] args) throws Exception {
+        // Logging is configured first before any references to Logger or LoggerFactory.
+        // Any public method or field accessors needed in IRI should be put in IRI and then delegate to IRILauncher. That
+        // ensures that future code does not need to know about this setup.
+        configureLogging();
+        IRILauncher.main(args);
+    }
+
+    private static void configureLogging() {
+        String config = System.getProperty("logback.configurationFile");
+        String level = System.getProperty("logging-level", "").toUpperCase();
+        switch (level) {
+            case "OFF":
+            case "ERROR":
+            case "WARN":
+            case "INFO":
+                break;
+            case "DEBUG":
+            case "TRACE":
+            case "ALL":
+                level = "DEBUG";
+                break;
+            default:
+                level = "INFO";
+        }
+        System.getProperties().put("logging-level", level);
+        System.out.println("Logging: property 'logging-level' set to : [" + level + "]");
+        System.out.println("Logging: levels can be one of: OFF, ERROR, WARN, INFO, DEBUG");
```

Comment:
Our tests fail because you print the word "ERROR". I wanted to fix the tests so that they don't fail. However after an internal discussion it was decided that the safest fix is for you to remove this line.

Reply:
I published it again with it all on lowercase - to see if it still triggers the fail.

Reply:
... And it did. Line now removed.